tests/qchem/test_tapering.py

```diff
@@ -629,3 +630,209 @@ def test_taper_obs(symbols, geometry, charge):
+            (1, -1, -1),
+            2,
+            4,
+            (0, 1),
+        ),
+        (
+            [
+                qml.Hamiltonian([1.0], [qml.PauliZ(0) @ qml.PauliZ(2)]),
+                qml.Hamiltonian([1.0], [qml.PauliZ(1) @ qml.PauliZ(3)]),
+            ],
+            [qml.PauliX(wires=[2]), qml.PauliX(wires=[3])],
+            (-1, -1),
+            2,
+            4,
+            (2, 1),
+        ),
+        (
+            [
+                qml.Hamiltonian([1.0], [qml.PauliZ(0) @ qml.PauliZ(2) @ qml.PauliZ(4)]),
+                qml.Hamiltonian([1.0], [qml.PauliZ(1) @ qml.PauliZ(3) @ qml.PauliZ(5)]),
+            ],
+            [qml.PauliX(wires=[4]), qml.PauliX(wires=[5])],
+            (-1, -1),
+            2,
+            6,
+            (4, 4),
+        ),
+        (
+            [
+                qml.Hamiltonian(
+                    [1.0], [qml.PauliZ(0) @ qml.PauliZ(1) @ qml.PauliZ(4) @ qml.PauliZ(5)]
+                ),
+                qml.Hamiltonian(
+                    [1.0], [qml.PauliZ(0) @ qml.PauliZ(2) @ qml.PauliZ(4) @ qml.PauliZ(6)]
+                ),
+                qml.Hamiltonian(
+                    [1.0], [qml.PauliZ(0) @ qml.PauliZ(3) @ qml.PauliZ(4) @ qml.PauliZ(7)]
+                ),
+            ],
+            [qml.PauliX(wires=[5]), qml.PauliX(wires=[6]), qml.PauliX(wires=[7])],
+            (1, 1, 1),
+            4,
+            8,
+            (4, 10),
+        ),
+    ],
+)
+def test_taper_excitations(
+    generators, paulixops, paulix_sector, num_electrons, num_wires, num_commuting
+):
+    r"""Test that the tapered excitation operators using :func:`~.taper_operation`
+    are consistent with the tapered Hartree-Fock state."""
+
+    hf_state = np.where(np.arange(num_wires) < num_electrons, 1, 0)
+    hf_tapered = taper_hf(generators, paulixops, paulix_sector, num_electrons, num_wires)
+    particle_num = qml.qchem.particle_number(num_wires)
+    particle_num_tapered = qml.taper(particle_num, generators, paulixops, paulix_sector)
+
+    o = np.array([1, 0])
+    l = np.array([0, 1])
+    state = functools.reduce(lambda i, j: np.kron(i, j), [l if s else o for s in hf_state])
+    state_tapered = functools.reduce(
+        lambda i, j: np.kron(i, j), [l if s else o for s in hf_tapered]
+    )
+    singles, doubles = qml.qchem.excitations(num_electrons, num_wires)
+
+    singles_obs = [qml.SingleExcitation(1, wires=single) for single in singles]
+    doubles_obs = [qml.DoubleExcitation(1, wires=double) for double in doubles]
+
+    singles_tap = [
+        taper_operation(op, generators, paulixops, paulix_sector, range(num_wires))
+        for op in singles_obs
+    ]
+    doubles_tap = [
+        taper_operation(op, generators, paulixops, paulix_sector, range(num_wires))
+        for op in doubles_obs
+    ]
+
+    singles_tap = [x for x in singles_tap if x]
+    doubles_tap = [x for x in doubles_tap if x]
```

Comment:
This looks like it can be further simplified by wrapping the logic in a for loop and iterating over `[singles, doubles ]`